src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs

```diff
@@ -86,45 +88,64 @@ public sealed class CosmosFhirOperationDataStore : IFhirOperationDataStore
 
         private string CollectionId { get; }
 
-        public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
+        public override Task<ExportJobOutcome> GetExportJobByIdAsync(string id, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
-
-            var hashObject = new { jobRecord.RequestUri, jobRecord.RequestorClaims };
-            var hash = JsonConvert.SerializeObject(hashObject).ComputeHash();
-
-            var resultFromHash = await GetExportJobByHashAsync(hash, cancellationToken);
-            if (resultFromHash != null)
+            try
             {
-                return resultFromHash;
+                return base.GetExportJobByIdAsync(id, cancellationToken);
             }
+            catch (JobNotFoundException)
```

Comment:
When a job isn't found Cosmos is throwing an aggregate exception that is wrapping the job not found exception. This isn't being caught right now.

Reply:
Hopefully this is fixed. Headpalm moment on the task await